pkg/datapath/link/link.go

```diff
@@ -16,12 +16,25 @@ package link
 
 import (
 	"fmt"
+	"time"
 
+	"github.com/cilium/cilium/pkg/lock"
 	"github.com/cilium/cilium/pkg/mac"
 
 	"github.com/vishvananda/netlink"
 )
 
+var linkCache ifNameCache
+
+func init() {
+	go func() {
+		for {
+			linkCache.syncCache()
```

Comment:
Since calling this involves a netlink call behind the scenes, what would be the intended failure mode? This will fail silently and cause the link cache to always be empty. Some form of error handling is likely required here.

I just noticed this was copied from `ifindex.go`, where errors also used to be silently dropped. In most cases, this call will fail due to permissions, so IMO this can exit or log error/crit.

Reply:
I don't think a failure here is critical enough to warrant killing the Cilium agent, but a logging the error and exiting the go routine makes a ton of sense, will add it.